src/utilities.cpp

```diff
@@ -45,6 +45,21 @@ Eigen::VectorXd DiscreteDifferentiation(const Eigen::VectorXd& input_vector, dou
   return derivative;
 };
 
+Eigen::VectorXd normalize(const Eigen::VectorXd& x)
+{
+  const double min = x.minCoeff();
+  const double max = x.maxCoeff();
+  size_t num_points = x.size();
+  Eigen::VectorXd x_norm = Eigen::RowVectorXd::Zero(num_points);
```

Comment:
Do you think it should be like this? I guess what you have works, but it seems like the type should match.

```suggestion
  Eigen::RowVectorXd x_norm = Eigen::RowVectorXd::Zero(num_points);
```